TradingBot/Services/MarketDataSource.cs

```diff
@@ -73,12 +80,14 @@ public async ValueTask DisposeAsync()
         _logger.Debug("Getting prices for all symbols from {Start} to {End}", start, end);
 
         var valid = await GetValidSymbolsAsync(slice ?? _tradingTask.SymbolSlice, token);
+        var fearGreedIndexes = await GetFearGreedIndexData(start.ToString("yyyy-MM-dd"), end.ToString("yyyy-MM-dd"), token);
+        var tradedValues = await GetTradedValueData(start, end, token);
         return await valid.Chunk(50)
             .ToAsyncEnumerable()
             .SelectManyAwait(async chunk =>
             {
                 _logger.Verbose("Getting data for chunk: {Symbols}", chunk.Select(t => t.Value).ToList());
-                var dataForChunk = await Task.WhenAll(chunk.Select(s => GetSymbolDataAsync(s, start, end, token)));
+                var dataForChunk = await Task.WhenAll(chunk.Select(s => GetSymbolDataAsync(s, start, end, fearGreedIndexes, tradedValues, token)));
                 return dataForChunk.ToAsyncEnumerable();
```

Comment:
We could return some other type from `GetSymbolDataAsync()` (which would basically contain what `DailyTradingData` contained earlier), and then add traded values and fear/greed indexes in another `.Select()`. This way we wouldn't have to pass lists through 5 different levels of abstraction. While we're at it, we could have date-indexed dictionaries instead of lists.